cached_proc_macro/src/cached.rs

```diff
@@ -249,16 +251,38 @@ pub fn cached(args: TokenStream, input: TokenStream) -> TokenStream {
             #set_cache_and_return
         }
     } else {
-        quote! {
-            {
+        if args.result_fallback {
+            quote! {
+                let old_val = {
+                    #lock
+                    let (result, has_expired) = cache.cache_get_expired(&key);
+                    if let (Some(result), false) = (result, has_expired) {
+                        #return_cache_block
+                    }
+                    result
+                };
+                #function_call
                 #lock
-                if let Some(result) = cache.cache_get(&key) {
-                    #return_cache_block
+                let result = match (result.is_err(), old_val) {
+                    (true, Some(result)) => {
+                        Ok(result)
```

Comment:
Should this be Ok(old_val)?

Reply:
i used shadowing :) but you're right, i'll rename this to be clearer